### ISSUE TITLE
CSV file autoformat error

### ISSUE THREAD

**Issue by hydrastarmaster** [biolab/orange3#6052]
In the "CSV File import" widget, the automatic type assigned to numbers truncates values to N e+16.
In this example, it's also strange that the column _"Feat2"_ is shown as descriptive after the "Group by" action. Interposing a "Select column" does not give any improvement.

Here attached, the files to reproduce the error.
[Orange - Check 6052.zip](https://github.com/biolab/orange3/files/9038539/Orange.-.Check.6052.zip)

**Comment by hydrastarmaster** [comment]
Windows 10 current release, Orange 3.32.0.
Here are some screenshots to show you how I see data (following the files I've attached in the first post).

1) RE: Autoformat
Original data (CSV as viewed by Excel; in this case w/ decimal separator ","):
![image](https://user-images.githubusercontent.com/41840980/177361537-9c0ddca9-2a4d-4060-a7d2-1fc67812d923.png)
Apply this:
![image](https://user-images.githubusercontent.com/41840980/177361594-905b311c-67bb-4b5b-b7a8-10877352d2b4.png)
and inside the widget (and in the following data table) there is:
![image](https://user-images.githubusercontent.com/41840980/177361758-02f310e5-f894-4d65-b136-02da83007bad.png)
As you can see, the E+ is truncated to 16. This is due to the automatic assignation [whilst it works well when numeric type is manually assigned].

2) RE: "Group by" meta anomaly
Having this:
![image](https://user-images.githubusercontent.com/41840980/177362677-1fefa6e9-dad8-483c-ba40-516b9036b518.png)
Input table:
![image](https://user-images.githubusercontent.com/41840980/177362713-bac93e0b-7d9e-47e2-801d-523f62641aaf.png)
Grouped this way:
![image](https://user-images.githubusercontent.com/41840980/177362750-e5be3e3a-0b75-4940-a32a-09e44c6d9be2.png)
Output table:
![image](https://user-images.githubusercontent.com/41840980/177362858-bbc3774b-9da9-4130-a057-b2f95bb480c4.png)
And "Feat2" should not be (also) a meta-feature (otherwise, please, explain why).

RE: https://orange3.readthedocs.io/projects/orange-visual-programming/en/latest/widgets/data/groupby.html
The example shows an "Iris" feature without Aggregations applied (that's the main field of the aggregation, obviously). But in the "Group by" widget, unchecking the "Concatenate" makes "Feat1" disappear. I suppose the difference is that "Iris" is defined as Class, while "Feat1" is automatically recognized as a text feature (... pretend it contains license plate numbers ...), and the widget treats them differently (maybe in an unexpected way).
The "Select column" applied before the "Group by" does not bring any change.

**Comment by hydrastarmaster** [comment]
1) You're right... Sorry. Well, thus the problem is with Excel and the decimal separator ... I saw a mirage earlier. Thanks.

2) I've changed just the type of "Feat1" to Class type (from Text):
![image](https://user-images.githubusercontent.com/41840980/177420306-40e96100-32b5-4776-98ea-b14ea84b5d43.png)
Then in "Group by" the sequence has changed, and "Feat1" appears as the first row and works like in the Iris example, without any Aggregation tick:
![image](https://user-images.githubusercontent.com/41840980/177420600-428cf300-5b0e-4ff4-96fe-dcfb1b1ec34f.png)
And it comes out right:
![image](https://user-images.githubusercontent.com/41840980/177420689-4bb55959-3ec4-4f21-a403-7b55db9a5f88.png)
Thus it is something to do with Class/Text differences.  In any case, it seems to be manageable by the user.
